### PR TITLE
NEUBURG-7: Add flow type checking

### PR DESCRIPTION
**Pull request by NKlug**
* Added flow to every file.
* Added types for every warning flow returned for those files.

### CODE REVIEW

**Comment by michael-markl** [comment]
Do we need to add `@flow` to every file?
Can we use .flowconfig to set it up globally?
https://flow.org/en/docs/config/

**Comment by michael-markl** [comment]
Plus, we could think about adding https://github.com/gajus/eslint-plugin-flowtype to force type annotations. For example there are still many functions without return type... 🤔 
What do you think?

**Comment by NKlug** [comment]
In order not to have `@flow` in every file, we can add
 `[options]  all = true` 
to the `.flowconfig` and exclude the ones we don't want to have checked, like node_modules, dist, etc.
Concerning your second comment:
What I did so far is: Add `@flow` to a file, run flow and fix everything that popped up. I think everything that is not annotated yet is automatically set by flow, but I'm not sure. If we really want to have the maximum number of annotations, I think the plugin you suggested is a good option.

**Comment by michael-markl** [comment]
@NKlug I would suggest, we do both the .flowconfig and the eslint plugin, since nobody said something against it. Unless you are against it ofc :D

**Comment by maxammann** [comment]
If you start a project from scratch with flow in mind you can set all to true :) :+1: for that.

File: package.json [hunk 1]
```diff
@@ -32,13 +32,15 @@
     "babel-core": "^6.26.0",
     "babel-eslint": "^8.2.2",
     "babel-preset-env": "^1.6.1",
+    "babel-preset-flow": "^6.23.0",
     "es6-template-string-loader": "^2.0.0",
```
Comment (maxammann) [inline package.json]: Can we remove this plugin? Its for webpack

**Review by maxammann — APPROVED** [review]

**Comment by maxammann** [comment]
Let's do the eslint-plugin-flowtype in a separate task though.

**Comment by NKlug** [comment]
I'm currently trying to set `all=true`, but I'm facing some issues:
Flow does check every file, even those without annotation, including the whole 'node_modules' folder, which is quite large and this process causes my laptop to crash :(
So what I tried is ignoring the node_modules folder in the `.flowconfig`, but then I get some flow errors because it can't find specific modules (because it's ignoring them). I haven't found a workaround for this yet and it seems like others are also struggling with the same issue.
At the moment I suggest removing the `all=true` again and writing the `@flow` tag in every file.
In the eslint-plugin we could then enable forcing said tag so we won't forget it anywhere.

@Schedulaar @maxammann What's your opinion on this and do you by chance know a solution to the problem?

File: package.json (hunk 1 above)
Comment (NKlug) [reply package.json]: Done

**Comment by NKlug** [comment]
Also see [this](https://github.com/facebook/flow/issues/3728) and [this](https://github.com/facebook/flow/issues/869). 
I decided to remove `all=true` again.

**Pull request merged**